### ISSUE TITLE
Unable to build

### ISSUE THREAD

**Issue by crnormand** [richardwilkes/gcs#101]
The build instructions indicate that JDK 10.   However, when I try (with JDK10.0.2), it reports:

C:\E\GIT_CODE\gcs>ant bundle
Buildfile: C:\E\GIT_CODE\gcs\build.xml

deps:

build:
    [mkdir] Created dir: C:\E\GIT_CODE\org.apache.commons.logging\antbuild
    [javac] Compiling 5 source files to C:\E\GIT_CODE\org.apache.commons.logging\antbuild
    [javac] javac: invalid target release: 11
    [javac] Usage: javac <options> <source files>
    [javac] use --help for a list of possible options

BUILD FAILED
C:\E\GIT_CODE\gcs\build.xml:96: The following error occurred while executing this line:
C:\E\GIT_CODE\org.apache.commons.logging\build.xml:21: Compile failed; see the compiler error output for details.

Total time: 0 seconds

When I try to build using JDK11.0.3, it reports:

C:\E\GIT_CODE\gcs\build.xml:125: Execute failed: java.io.IOException: Cannot run program "javapackager" (in directory "C:\E\GIT_CODE
\gcs"): CreateProcess error=2, The system cannot find the file specified
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
        at java.base/java.lang.Runtime.exec(Runtime.java:591)
        at org.apache.tools.ant.taskdefs.launcher.Java13CommandLauncher.exec(Java13CommandLauncher.java:58)

It seems that the "javapackager" was removed in JDK11.

So I tried compiling with JDK11, but including JDK10 at the end of the path, so that it could find the javapackager application.    That got the farthest:

C:\E\GIT_CODE\gcs>ant bundle
Buildfile: C:\E\GIT_CODE\gcs\build.xml

deps:

build:
    [mkdir] Created dir: C:\E\GIT_CODE\org.apache.commons.logging\antbuild
    [javac] Compiling 5 source files to C:\E\GIT_CODE\org.apache.commons.logging\antbuild
   [delete] Deleting directory C:\E\GIT_CODE\org.apache.commons.logging\antbuild
      [zip] Building zip: C:\E\GIT_CODE\java_modules\org.apache.commons.logging-1.2-src.zip

build:
    [mkdir] Created dir: C:\E\GIT_CODE\org.apache.fontbox\antbuild
    [javac] Compiling 103 source files to C:\E\GIT_CODE\org.apache.fontbox\antbuild
   [delete] Deleting directory C:\E\GIT_CODE\org.apache.fontbox\antbuild
      [zip] Building zip: C:\E\GIT_CODE\java_modules\org.apache.fontbox-2.0.9-src.zip

build:
    [mkdir] Created dir: C:\E\GIT_CODE\org.apache.pdfbox\antbuild
    [javac] Compiling 562 source files to C:\E\GIT_CODE\org.apache.pdfbox\antbuild
   [delete] Deleting directory C:\E\GIT_CODE\org.apache.pdfbox\antbuild
      [zip] Building zip: C:\E\GIT_CODE\java_modules\org.apache.pdfbox-2.0.9-src.zip

build:
    [mkdir] Created dir: C:\E\GIT_CODE\gnu.trove\antbuild
    [javac] Compiling 692 source files to C:\E\GIT_CODE\gnu.trove\antbuild
   [delete] Deleting directory C:\E\GIT_CODE\gnu.trove\antbuild
      [zip] Building zip: C:\E\GIT_CODE\java_modules\gnu.trove-3.0.3-src.zip

build:
    [mkdir] Created dir: C:\E\GIT_CODE\com.lowagie.text\antbuild
    [javac] Compiling 343 source files to C:\E\GIT_CODE\com.lowagie.text\antbuild
   [delete] Deleting directory C:\E\GIT_CODE\com.lowagie.text\antbuild
      [zip] Building zip: C:\E\GIT_CODE\java_modules\com.lowagie.text-2.1.7-src.zip

build:
    [mkdir] Created dir: C:\E\GIT_CODE\toolkit\antbuild
    [javac] Compiling 410 source files to C:\E\GIT_CODE\toolkit\antbuild
   [delete] Deleting directory C:\E\GIT_CODE\toolkit\antbuild
      [zip] Building zip: C:\E\GIT_CODE\java_modules\com.trollworks.toolkit-4.12.1-src.zip

build:
   [delete] Deleting directory C:\E\GIT_CODE\gcs\antbuild
    [mkdir] Created dir: C:\E\GIT_CODE\gcs\antbuild
    [javac] Compiling 250 source files to C:\E\GIT_CODE\gcs\antbuild
   [delete] Deleting directory C:\E\GIT_CODE\gcs\antbuild
   [delete] Deleting: C:\E\GIT_CODE\gcs\manifest
      [zip] Building zip: C:\E\GIT_CODE\java_modules\com.trollworks.gcs-4.11.1-src.zip

bundle:
    [mkdir] Created dir: C:\E\GIT_CODE\gcs\antbuild\icons
     [exec] Generating icons into antbuild\icons
     [exec] Created: antbuild\icons\app.ico
     [exec] Created: antbuild\icons\adq.ico
     [exec] Created: antbuild\icons\eqp.ico
     [exec] Created: antbuild\icons\gcs.ico
     [exec] Created: antbuild\icons\gct.ico
     [exec] Created: antbuild\icons\skl.ico
     [exec] Created: antbuild\icons\spl.ico
     [exec] Created: antbuild\icons\not.ico
     [exec] Created: antbuild\icons\pdf.ico
    [mkdir] Created dir: C:\E\GIT_CODE\gcs\antbuild\gcs-4.11.1
     [exec] Creating app bundle: gcs in C:\E\GIT_CODE\gcs\antbuild\gcs-4.11.1
     [exec] Module com.trollworks.gcs does not exist.
     [exec] "Adding modules: [com.trollworks.gcs] to runtime image."
     [exec] Exception: java.lang.module.FindException: Error reading module: ..\java_modules\com.lowagie.text-2.1.7.jar
     [exec] Error: Bundler "Windows Application Image" (windows.app) failed to produce a bundle.
     [exec] Result: -1
     [move] Moved 3 empty directories to 2 empty directories under C:\E\GIT_CODE\gcs\antbuild\gcs-4.11.1
    [mkdir] Created dir: C:\E\GIT_CODE\gcs\antbuild\gcs-4.11.1\support
     [copy] Copying 9 files to C:\E\GIT_CODE\gcs\antbuild\gcs-4.11.1\support
     [copy] Copying 998 files to C:\E\GIT_CODE\gcs\antbuild\gcs-4.11.1\Library
   [delete] Deleting: C:\E\GIT_CODE\gcs\gcs-4.11.1.zip
      [zip] Building zip: C:\E\GIT_CODE\gcs\gcs-4.11.1.zip
   [delete] Deleting directory C:\E\GIT_CODE\gcs\antbuild

BUILD SUCCESSFUL
Total time: 1 minute 2 seconds

Any thoughts?

**Comment by richardwilkes** [comment]
You need to grab the source for the last release to build with Java 10. The current HEAD has been moved up to more recent versions, but that comes with the problem of the Packager app having been dropped without a replacement. I started addressing this, but don’t have a lot of spare time for GCS at the moment and have not completed the work yet.

**Comment by crnormand** [comment]
Ok, no worries.   I can wait.   I was able to get everything into Eclipse, and I have been able to run from there.   I am working on an output template to create the XML needed to import GCS characters into Fantasy Grounds.  https://www.fantasygrounds.com/forums/showthread.php?36043-GURPS-4E-Core-Ruleset .   

It requires a few modifications to the TextTemplate class.  Nothing earth shattering, but the author of the GURPS ruleset is very interested in it, since it will open up a new way to import characters from a new (and dare I say, better) character editor ;-)
